dsb-gui/src/main/java/com/spectralogic/dsbrowser/gui/components/ds3panel/ds3treetable/Ds3TreeTablePresenter.java

```diff
@@ -214,41 +220,56 @@ protected Object call() throws Exception {
                                     ds3PanelPresenter.filterChanged(ds3PanelPresenter.getSearchedText());
                                 }
                                 // deepStorageBrmainowserPresenter.logText("Delete response code: " + deleteObjectsResponse.getStatusCode(), LogType.SUCCESS);
+                                if (ds3TreeTable.getRoot() == null || ds3TreeTable.getRoot().getValue() == null) {
+                                    ds3TreeTable.setRoot(ds3TreeTable.getRoot().getParent());
+                                    Platform.runLater(() -> {
+                                        ds3TreeTable.getSelectionModel().clearSelection();
+                                        ds3PanelPresenter.getDs3PathIndicator().setText("");
+                                        ds3PanelPresenter.getDs3PathIndicatorTooltip().setText("");
+                                    });
+
+                                } else {
+                                    ds3TreeTable.setRoot(selectedItem);
+                                }
+                                ds3TreeTable.getSelectionModel().select(selectedItem);
+                                RefreshCompleteViewWorker.refreshCompleteTreeTableView(ds3Common, workers);
                                 deepStorageBrowserPresenter.logText("Successfully deleted file(s)", LogType.SUCCESS);
+
                             });
                         }
                     }
 
-                } catch (final IOException e) {
-                    if (e instanceof FailedRequestException) {
-                        LOG.error("Failed to delete files" + e);
-                        Platform.runLater(() -> deepStorageBrowserPresenter.logText("Failed to delete files : " + ((FailedRequestException) e).getError().getMessage(), LogType.ERROR));
-                        ALERT.setContentText("Failed to delete a files");
-                        ALERT.showAndWait();
-                    } else {
-                        LOG.error("Failed to delete files" + e);
-                        Platform.runLater(() -> deepStorageBrowserPresenter.logText("Failed to delete files", LogType.ERROR));
-                        ALERT.setContentText("Failed to delete a files");
-                        ALERT.showAndWait();
-                    }
+                } catch (final FailedRequestException fre) {
```

Comment:
If this exception catch block is doing the same logic ast the IOException catch block bellow it, remove this catch statement, and rely on the IOException catch block